contracts/staking/test/stake_test.ts

```diff
@@ -1,47 +1,357 @@
 import { ERC20ProxyContract, ERC20Wrapper } from '@0x/contracts-asset-proxy';
```

Comment:
(File Wide Comment) These tests are amazingly readable, and after reading through the `StakerActor`, they are also rigorous 😄. The actor pattern is sweet!